Display version

charleygros opened the pull request:
This PR fetches the `ivadomed` version, either release version or git commit, and display it in the terminal each time a script is launched.

Note: adding the file `ivadomed/version.txt` were is indicated the release version, instead of `ivadomed/__init__.py`.

Fixes #455.

charleygros committed:
main: rm path_preds as evaluate argument

charleygros committed:
utils: version dir

charleygros committed:
utils: _version_string

charleygros committed:
utils: _git_info

charleygros committed:
utils: check_exe

charleygros committed:
utils: __get_commit

charleygros committed:
utils: __get_branch

charleygros committed:
utils: move things around

charleygros committed:
__init__: version and dir

charleygros committed:
version.txt

charleygros committed:
setup.py: release version

charleygros committed:
utils.py: get release version from version.txt

charleygros committed:
utils.py: add ivadomed in path

charleygros committed:
utils.py: init_ivadomed

charleygros committed:
main.py: init_ivadomed

charleygros committed:
scripts: init_ivadomed

charleygros committed:
utils: cleanup

charleygros committed:
aesthetics

charleygros committed:
new line

charleygros committed:
fix merge

charleygros committed:
rm dvp lines

charleygros committed:
silly me

charleygros commented:
Message at the top of each scripts:
```
--
IVADOMED (git-cg/version-b26271a0bd80695b2c3ba758b5fd70888fe5db7c)

Log directory already exists: spineGeneric
Using GPU number 0
```

charleygros commented:
TODO before merging: change the docstrings of new functions (thanks heaps to SCT team eheh)

jcohenadad commented on ivadomed/__init__.py:
```diff
@@ -1 +1,2 @@
-__version__ = "2.2.1"
```
is this change still compatible with RTD build (which fetches the version, i think..?)

Drulex replied:
Looks ok https://readthedocs.org/projects/ivado-medical-imaging/builds/11961354/

lrouhier commented on ivadomed/scripts/convert_to_onnx.py:
```diff
@@ -50,4 +50,5 @@ def main():
 
 
 if __name__ == '__main__':
+    imed_utils.init_ivadomed()
```
I performed a test with this script and the version did not show.. would it be possible to put that directly in the main function ?

charleygros replied:
You are very correct! Thanks ++
Corrected, please let me know if the problem remains, cheers

charleygros committed:
scripts: move init

charleygros committed:
Merge branch 'cg/version' of https://github.com/neuropoly/ivado-medical-imaging into cg/version

charleygros committed:
utils: _git_info docstring

charleygros committed:
utils: check_exe docstring

charleygros committed:
utils: __get_commit docstring

charleygros commented:
> TODO before merging: change the docstrings of new functions (thanks heaps to SCT team eheh)

Done.

andreanne-lemay reviewed (approved):
Works well on my end :)! Thank you for this change, my only comment would be why the two "--", it is not super aesthetic, but really minor.
![image](https://user-images.githubusercontent.com/49137243/94383407-f31e8b80-010d-11eb-978f-5f93b4a02230.png)

jcohenadad commented:
how does everyone feel about displaying “ivadomed” (without capital letters), as it is the official writing of the software package?

charleygros committed:
utils: rm -- lower cases

charleygros commented:
> my only comment would be why the two "--", it is not super aesthetic, but really minor.

Yep, removed.

> how does everyone feel about displaying “ivadomed” (without capital letters), as it is the official writing of the software package?

Yep, agreed, corrected.
